python/ray/train/gbdt_trainer.py

```diff
@@ -181,12 +193,29 @@ def training_loop(self) -> None:
 
         config.setdefault("verbose_eval", False)
         config.setdefault("callbacks", [])
-        config["callbacks"] += [
-            self._tune_callback_cls(filename=MODEL_KEY, frequency=1)
-        ]
+
+        if not any(
+            isinstance(
+                cb, (self._tune_callback_report_cls, self._tune_callback_checkpoint_cls)
+            )
+            for cb in config["callbacks"]
+        ):
+            # Only add our own callback if it hasn't been added before
+            checkpoint_frequency = (
+                self.run_config.checkpoint_config.checkpoint_frequency
+            )
+            if checkpoint_frequency > 0:
```

Comment:
can we ban this in the future?

Reply:
Users can always add their own callbacks (which could also be non-reporting, or derived from our reporting callbacks), so we should check if the callbacks exist. We do the same thing in Tune. But for better readability I can at least put this into a separate function.